homeassistant/components/camera/__init__.py

```diff
@@ -761,22 +713,40 @@ async def async_refresh_providers(self) -> None:
 
         Returns True if any state was updated (and needs to be written)
         """
-        old_state = self._rtsp_to_webrtc
-        self._rtsp_to_webrtc = await self._async_use_rtsp_to_webrtc()
-        if old_state != self._rtsp_to_webrtc:
+        old_providers = self._webrtc_providers
+        self._webrtc_providers = await self._async_get_supported_webrtc_providers()
```

Comment:
I'd cache the providers in a local variable too since we need to access it twice.